### PR TITLE
Move log detail aggregation and reuse it in syslog/systemd logger

### PR DESCRIPTION
**Pull request by juliusknorr**
This way we also include details like the request id in the log when using different logging backends like syslog or systemd log.

### CODE REVIEW

File: lib/private/Log/File.php [hunk 1]
```diff
@@ -47,7 +47,7 @@
  * Log is saved at data/nextcloud.log (on default)
  */
 
-class File implements IWriter, IFileBased {
+class File extends LogDetails implements IWriter, IFileBased {
```
Comment (kesselb) [inline lib/private/Log/File.php]: Mind to add `LogDetails` and bump the autoloaders? :thinking:

Comment (MorrisJobke) [reply lib/private/Log/File.php]: 😆

Comment (MorrisJobke) [reply lib/private/Log/File.php]: The autoloaders are added, but the file is still missing. 🤔

Comment (juliushaertl) [reply lib/private/Log/File.php]: Oops, sorry :see_no_evil: 

Added it as well

**Review by rullzer — APPROVED** [review]
Looks sane

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Pull request merged**